File: Homework/HW1/hw1.cpp
#include <iostream>
#include <tgmath.h>
#include <ctime>

using namespace std;

float performCalculation(){
    //Create start/stop vals for caluclation
    float initialVal = 1.0, endVal = 2.0;
    
    //Create variables for sum and x
    float totalSum = 0.0, x = initialVal;
    
    //Iterate through all possible values for x from startVal to endVal
    while (x < endVal) {
        totalSum += (1/x);
        
        //Increase value of x with smallest possible value for float type
        x = nextafterf(x, endVal);
    }
    
    return totalSum;
}

int main(){
    clock_t startTime = clock();            //Start clock
    float sum = performCalculation();       //Perform calculation
    clock_t endTime = clock();              //End clock
    
    //Print results
    cout << "Sum: " << sum << endl;
    cout << "Total Time: " << (endTime - startTime)/ (double)(CLOCKS_PER_SEC / 1000) << "ms" << endl;
}

File: Homework/HW1/Submitted/hw1.py
import time
import numpy as np


def performSummation():
    # Instantiate summation range values
    start_val = np.float32(1.0)
    end_val = np.float32(2.0)

    sum = 0.0
    x = start_val
    # num_vals = 0

    while(x < end_val):
        sum += 1/x
        x = np.nextafter(x, end_val)
    # num_vals += 1
    return sum




def main():
    # Get start time of execution
    start_time = time.time()

    # Perform Summation calculations
    sum = performSummation()
    
    # Get end time of execution
    end_time = time.time()
    
    print("SUM: %s" % sum)
    
    # Calculate total execution time
    print("Execution Time: %s" % (end_time - start_time))


main()